lib/internal/main/test_runner.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+const { ArrayFrom, Promise, SafeSet } = primordials;
+const {
+  prepareMainThreadExecution
+} = require('internal/bootstrap/pre_execution');
+const { spawn } = require('child_process');
+const { readdirSync, statSync } = require('fs');
+const console = require('internal/console/global');
+const {
+  codes: {
+    ERR_TEST_FAILURE,
+  },
+} = require('internal/errors');
+const test = require('internal/test_runner/harness');
+const { kSubtestsFailed } = require('internal/test_runner/test');
+const { doesPathMatchFilter } = require('internal/test_runner/utils');
+const { join, resolve } = require('path');
+const kFilterArgs = ['--test'];
+
+prepareMainThreadExecution(false);
+markBootstrapComplete();
+
+// TODO(cjihrig): Replace this with recursive readdir once it lands.
+function processPath(path, testFiles, userSupplied = false) {
+  const stats = statSync(path);
+
+  if (stats.isFile()) {
+    if (userSupplied || doesPathMatchFilter(path)) {
+      testFiles.add(path);
+    }
+  } else if (stats.isDirectory()) {
+    if (!userSupplied && path.endsWith('node_modules')) {
+      return;
+    }
+
+    const entries = readdirSync(path);
+
+    for (let i = 0; i < entries.length; i++) {
+      processPath(join(path, entries[i]), testFiles);
+    }
+  }
+}
+
+function createTestFileList() {
+  const testPaths = process.argv.length > 1 ? process.argv.slice(1) : ['.'];
+  const testFiles = new SafeSet();
+
+  try {
+    for (let i = 0; i < testPaths.length; i++) {
+      const absolutePath = resolve(testPaths[i]);
+
+      processPath(absolutePath, testFiles, true);
+    }
+  } catch (err) {
+    if (err?.code === 'ENOENT') {
+      console.error(`Could not find '${err.path}'`);
+      process.exit(1);
+    }
+
+    throw err;
+  }
+
+  return ArrayFrom(testFiles).sort();
+}
+
+function runTestFile(path) {
+  return test(path, () => {
```

Comment:
```suggestion
  return test(path, async () => {
```
to guarantee no exceptions are thrown

Reply:
I would like to know if an exception happens since this file only runs core code. If there is an exception it means there is a bug in core.